numpy/core/tests/test_ufunc.py

```diff
@@ -1933,4 +1933,13 @@ def test_ufunc_noncontiguous(ufunc):
             warnings.filterwarnings("always")
             res_c = ufunc(*args_c)
             res_n = ufunc(*args_n)
-        assert_equal(res_c, res_n)
+        dt = np.find_common_type([np.dtype(c) for c in out], [])
```

Comment:
Hmmm, maybe 
```
if len(out) == 1:
    res_c = (res_c,)
    res_n = (res_n,)
```
and then, `for c, n in zip(res_c, res_n): ...` then you can avoid the whole common type magic?

Reply:
I tried to implement this suggestion, as well as the one to use `nulp` variant only to discover that it does not like `NaN` in the array:

```
>>> import numpy as np
>>> x = np.full((5,), np.nan, 'd')
>>> np.testing.assert_array_almost_equal_nulp(x, x, nulp=2)
/tmp/numpy_upstream/numpy/testing/_private/utils.py:1566: RuntimeWarning: invalid value encountered in greater
  ref = nulp * np.spacing(np.where(ax > ay, ax, ay))
/tmp/numpy_upstream/numpy/testing/_private/utils.py:1567: RuntimeWarning: invalid value encountered in less_equal
  if not np.all(np.abs(x-y) <= ref):
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/git/numpy_upstream/numpy/testing/_private/utils.py", line 1573, in assert_array_almost_equal_nulp
    raise AssertionError(msg)
AssertionError: X and Y are not equal to 2 ULP (max is 0)
```

So I will perhaps stick with `assert_allclose`

Reply:
Sounds like that function needs fixing, I'll open an issue. I'm fine with assert_allclose for now.

Reply:
Should we use `atol=0` though, or does that fail for denormals, or maybe `atol` should be just much smaller? Oh well, maybe just leave it in any case, right...